src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+import decimal
+import inspect
+import re
+import sys
+
+from functools import reduce
+from pathlib import Path
+from typing import Any, List, Optional
+
+import snowflake.snowpark._internal.proto.ast_pb2 as proto
+
+NoneType = type(None)
+PYTHON_TO_AST_CONST_MAPPINGS = {
+    NoneType: "null_val",
+    bool: "bool_val",
+    int: "int64_val",
+    float: "float64_val",
+    str: "string_val",
+    bytearray: "binary_val",
+    bytes: "binary_val",
+    decimal.Decimal: "big_decimal_val",
+    datetime.date: "date_val",
+    datetime.time: "python_time_val",
+    datetime.datetime: "python_timestamp_val",
+}
+def infer_const_ast(obj: Any, ast: proto.Expr) -> None:
+    """Infer the Const AST expression from obj, and populate the provided ast.Expr() instance"""
+    if obj is None:
+        fill_src_position(ast.null_val.src)
```

Comment:
should we not call `fill_src_position()` in other cases?

Reply:
For now this was just to be able to create the `NullVal` message. Being conservative with usage just to be safe, but it can be called on every message extending the `Expr` trait. Should I do so for all consts?

Reply:
let's call it on all consts. You prob. can get rid off the if here